pkg/util/workqueue/telemetry/telemetry.go

```diff
@@ -0,0 +1,229 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package telemetry provides implementation for the workqueue metrics provider
+package telemetry
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/DataDog/datadog-agent/pkg/telemetry"
+)
+
+var (
+	commonOpts = telemetry.Options{NoDoubleUnderscoreSep: true}
+)
+
+type gaugeWrapper struct {
+	telemetry.Gauge
+}
+
+// Inc implements the workqueue.GaugeMetric interface
+func (g gaugeWrapper) Inc() {
+	g.Gauge.Inc()
+}
+
+// Dec implements the workqueue.GaugeMetric interface
+func (g gaugeWrapper) Dec() {
+	g.Gauge.Dec()
+}
+
+// Set implements the workqueue.GaugeMetric interface
+func (g gaugeWrapper) Set(v float64) {
+	g.Gauge.Set(v)
+}
+
+type counterWrapper struct {
+	telemetry.Counter
+}
+
+// Inc implements the workqueue.CounterMetric interface
+func (c counterWrapper) Inc() {
+	c.Counter.Inc()
+}
+
+type histgramWrapper struct {
+	telemetry.Histogram
+}
+
+// Observer implements the workqueue.HistogramMetric interface
+func (l histgramWrapper) Observe(value float64) {
+	l.Histogram.Observe(value)
+}
+
+// QueueMetricsProvider is a workqueue.MetricsProvider that provides metrics for the SBOM queue
+type QueueMetricsProvider struct {
+	mutex   *sync.Mutex
```

Comment:
I think the mutex does not need to be a pointer

Reply:
If the mutex is not a pointer, then we will need to have all the `QueueMetricsProvider` methods as pointer receiver methods, else we will be copying the lock on each time. However, if the methods are defined as pointer receivers, then QueueMetricsProvider will not be implementing `workqueue.MetricsProvider` anymore, since the interface defines these methods as value receivers.

Reply:
I am not sure I understand why the lock will be copied.